modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -356,12 +362,16 @@ class DistributedQueryManager {
 
         private volatile Long rootFragmentId = null;
 
-        private InternalTransaction transaction;
+        private @Nullable InternalTransaction transaction;
```

Comment:
Maybe I don't fully understand, but in my picture transaction should present always, it could be implicite, explicite or transaction sended in fragment to remote node.

Reply:
My bad - it should be implemented in separate ticket